benchmarks/bioid_ner_benchmark.py

```diff
@@ -0,0 +1,520 @@
+        output = {curie}
+        prefix, identifier = curie.split(':', maxsplit=1)
+        for xref_prefix, xref_id in bio_ontology.get_mappings(prefix,
+                                                              identifier):
+            output.add(f'{xref_prefix}:{xref_id}')
+
+        # TODO these should all be in bioontology, eventually
+        for xref_curie in self.equivalences.get(curie, []):
+            if xref_curie in output:
+                continue
+            xref_prefix, xref_id = xref_curie.split(':', maxsplit=1)
+            if (prefix, xref_prefix) not in BO_MISSING_XREFS:
+                BO_MISSING_XREFS.add((prefix, xref_prefix))
+                tqdm.write(
+                    f'Bioontology v{bio_ontology.version} is missing mappings'
+                    f' from {prefix} to {xref_prefix}')
+            output.add(xref_curie)
+
+        if prefix == 'NCBI gene':
+            hgnc_id = get_hgnc_from_entrez(identifier)
+            if hgnc_id is not None:
+                output.add(f'HGNC:{hgnc_id}')
+        if prefix == 'UP':
+            hgnc_id = get_hgnc_id(identifier)
+            if hgnc_id is not None:
+                output.add(f'HGNC:{hgnc_id}')
+        if prefix == 'PubChem':
+            chebi_id = get_chebi_id_from_pubchem(identifier)
+            if chebi_id is not None:
+                output.add(f'CHEBI:{chebi_id}')
+        return output
+
+    def _get_entity_type_helper(self, row) -> str:
+        if self._get_entity_type(row.obj) != 'Gene':
+            return self._get_entity_type(row.obj)
+        elif any(y.startswith('HGNC') for y in row.obj_synonyms):
+            return 'Human Gene'
+        else:
+            return 'Nonhuman Gene'
+
+    @staticmethod
+    def _get_entity_type(groundings: Collection[str]) -> str:
+        """Get entity type based on entity groundings of text in corpus."""
+        if any(
+                grounding.startswith('NCBI gene') or grounding.startswith('UP')
+                for grounding in groundings
+        ):
+            return 'Gene'
+        elif any(grounding.startswith('Rfam') for grounding in groundings):
+            return 'miRNA'
+        elif any(
+                grounding.startswith('CHEBI') or grounding.startswith('PubChem')
+                for grounding in groundings):
+            return 'Small Molecule'
+        elif any(grounding.startswith('GO') for grounding in groundings):
+            return 'Cellular Component'
+        elif any(
+                grounding.startswith('CVCL') or grounding.startswith('CL')
+                for grounding in groundings
+        ):
+            return 'Cell types/Cell lines'
+        elif any(grounding.startswith('UBERON') for grounding in groundings):
+            return 'Tissue/Organ'
+        elif any(
+                grounding.startswith('NCBI taxon') for grounding in groundings):
+            return 'Taxon'
+        else:
+            return 'unknown'
+
+    def _process_annotations_table(self):
```

Comment:
This is exactly the same as the corresponding method in the original benchmark class. So I would suggest either (1) making this benchmarker class a subclass of the original one or (2) introducing a parent class from which both benchmarker classes inherit.

Reply:
This is actually different from the original so we are keeping it